scratchapi2/front.py

```diff
@@ -16,93 +16,115 @@ class FrontPage
 from .user import Project, User
 from .gclass import GenericData
 
+GETINFO = False
+
 class FrontPage(object):
-    """ A class to handle front page """
+    """The Front Page of the Scratch website."""
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """Enforce this class as a singleton."""
+        if not cls._instance:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self, api_url="https://api.scratch.mit.edu/"):
+        """Initialize the object with the API URL."""
+        self.api_url = api_url
 
     def __repr__(self):
-        """ Always return <ScratchAPI2 FrontPage> """
-        return "<ScratchAPI2 FrontPage>"
+        """Represent the FrontPage."""
+        return "<SFrontPage>"
 
     __str__ = __repr__
 
-    @staticmethod
-    def news(limit=3, offset=0):
-        """ Get Scratch News """
-        result = requests.get("https://api.scratch.mit.edu/news?limit={0}&offset={1}".format(limit, offset)).json()
-        return_values = []
+    def _request(self, path, *opts, api_url=None):

```

Comment:
This is the `_request` method I suggested before. Notice the `api_url` keyword-only argument - it cannot be specified as a positional argument.